tgui/packages/tgui/interfaces/GeneModder.js

```diff
@@ -0,0 +1,262 @@
+import { useBackend } from "../backend";
+import {
+  Button,
+  Section,
+  Flex,
+  Icon,
+  Collapsible,
+  LabeledList } from "../components";
+import {
+  ComplexModal,
+} from '../interfaces/common/ComplexModal';
+import { Window } from "../layouts";
+
+export const GeneModder = (props, context) => {
+  const { data } = useBackend(context);
+  const { has_seed } = data;
+
+  return (
+    <Window resizable>
+      <Window.Content display="flex" className="Layout__content--flexColumn">
+        <Storage />
+        <ComplexModal maxWidth="75%" maxHeight="75%" />
+        {has_seed === 0 ? (
+          <MissingSeed />
+        ) : (
+          <Genes />
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const Genes = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { disk } = data;
+
+  return (
+    <Section
+      title="Genes"
+      flexGrow="1"
+      buttons={
+        <Button
+          content = "Insert Gene from Disk"
+          disabled = {!disk || !disk.can_insert || disk.is_core}
+          icon = "arrow-circle-down"
+          onClick={() => act('insert')}
+        />
+      }>
+      <CoreGenes/>
+      <ReagentGenes/>
+      <TraitGenes/>
+    </Section>
+  );
+};
+
+const MissingSeed = (props, context) => {
+  return (
+    <Section flexGrow="1">
+      <Flex height="100%">
+        <Flex.Item
+          bold
+          grow="1"
+          textAlign="center"
+          align="center"
+          color="green">
+          <Icon
+            name="leaf"
+            size={5}
+            mb="10px"
+          /><br />
+          The plant DNA manipulator is missing a seed.
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+};
+
+const Storage = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    has_seed, seed,
+    has_disk, disk
```

Comment:
```suggestion
    has_seed,
    seed,
    has_disk,
    disk
```
should keep it to 1 data const per line